browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -287,7 +287,8 @@ void SpeedreaderTabHelper::ProcessNavigation(
   auto* nav_entry = navigation_handle->GetNavigationEntry();
 
   const bool url_looks_readable =
-      nav_entry && !nav_entry->IsViewSourceMode() && rewriter_service &&
+      rewriter_service && nav_entry &&
+      nav_entry->GetVirtualURL().SchemeIsHTTPOrHTTPS() &&
```

Comment:
It prevents showing the button on viewsoure, ipfs and other pages.

Reply:
would be nice to have a test